test/gui/shared/steps/steps.py

```diff
@@ -336,17 +332,20 @@ def step(context):
 
 @When('the user waits for file "|any|" to be synced')
 def step(context, fileName):
-    waitForFileToBeSynced(context, fileName)
+    resource = join(context.userData['currentUserSyncPath'], fileName)
+    waitForFileToBeSynced(context, resource)
 
 
 @When('the user waits for folder "|any|" to be synced')
 def step(context, folderName):
-    waitForFolderToBeSynced(context, folderName)
+    resource = join(context.userData['currentUserSyncPath'], folderName)
+    waitForFolderToBeSynced(context, resource)
 
 
 @Given('the user has waited for file "|any|" to be synced')
 def step(context, fileName):
-    waitForFileToBeSynced(context, fileName)
+    resource = join(context.userData['currentUserSyncPath'], fileName)
+    waitForFileToBeSynced(context, resource)
```

Comment:
The logic here is same as `when the user waits for file "|any|" to be synced`. So can you reuse the logic? (By putting the logic in a different function and then calling the function from step definition)